resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -755,6 +755,7 @@ private[spark] class Client(
       props.store(writer, "Spark configuration.")
       writer.flush()
       confStream.closeEntry()
+      writer.close()
```

Comment:
You don't want this, because that will close the underlying stream. That's already closed below.

So if later another block is added to add another entry to the file, it will fail because the stream is closed.